### PR TITLE
[docs] Add RELEASES.md

### PR DESCRIPTION
**Pull request by bmorelli25**
First attempt at documenting the release process. Closes https://github.com/elastic/apm-server/issues/2197.

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
+# APM Server release checklist
+
+## Before the release
+
+* Create a PR that:
+    1. Uncomments out relevant release notes. Release notes are in the [changelogs](https://github.com/elastic/apm-server/tree/master/changelogs).
+    2. Updates the `:stack-version:`
+    * [Sample PR](https://github.com/elastic/apm-server/pull/2064/files)
+
+* Update `vendor/github.com/elastic/beats/libbeat/version/version.go`
```
Comment (graphaelli) [inline RELEASES.md]:
```suggestion
* Create a separate PR to update `vendor/github.com/elastic/beats/libbeat/version/version.go`
```

File: RELEASES.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    * [Sample PR](https://github.com/elastic/apm-server/pull/1886)
+
+* The following may also need to be updated manually:
+    * APM Overview's [release highlights](https://github.com/elastic/apm-server/blob/master/docs/guide/apm-release-notes.asciidoc) - Anything exciting across the APM stack!
+    * APM Overview's [breaking changes](https://github.com/elastic/apm-server/blob/master/docs/guide/apm-breaking-changes.asciidoc) - Any breaking changes across the APM stack.
+    * APM Server's [breaking changes](https://github.com/elastic/apm-server/blob/master/docs/breaking-changes.asciidoc) - Any APM Server breaking changes.
+    * APM Server's [upgrade guide](https://github.com/elastic/apm-server/blob/master/docs/upgrading.asciidoc).
+
+* Script for determining if changelogs are synced: https://gist.github.com/graphaelli/92fd6b5ff08d69600e880df2a234ac35
```
Comment (graphaelli) [inline RELEASES.md]: feel free to check this into the this repo as part of this change

Comment (bmorelli25) [reply RELEASES.md]: Ah, yeah, much better idea.

**Comment by bmorelli25** [comment]
Added both of your suggestions, thank you!

File: RELEASES.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    * Run the [`check_changelogs.py`](script/check_changelogs.py) script to ensure changelogs are synced across branches. This will soon be a PR check.
+        * Don't forget to update the "VERSIONS" to include a new branch if necessary.
+
+## On release day
+
+* New branches need to be added to `conf.yml` in the `elastic/docs` repo. [Example](https://github.com/elastic/docs/pull/893/files#diff-4a701a5adb4359c6abf9b8e1cb38819fR925). **This is handled by the docs release manager.**
+
+* Merge the above PRs
+
+* Push [tags](https://github.com/elastic/apm-server/releases) on GitHub.
```
Comment (graphaelli) [inline RELEASES.md]: this is now done automatically, let's convert this to a verification step instead

**Comment by bmorelli25** [comment]
Jenkins retest this

**Comment by bmorelli25** [comment]
I think this is good to go for now. Merging

**Pull request merged**